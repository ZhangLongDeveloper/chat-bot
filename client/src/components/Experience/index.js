import React from 'react';

const Experience = () => (
  <div>
    This is EXPERIENCE
  </div>
);

export default Experience;
