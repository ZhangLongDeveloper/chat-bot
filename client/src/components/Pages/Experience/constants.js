export const experiences = [
  {
    image: 'image 1',
    companyName: 'Cornerjob',
    companyLink: 'https://www.cornerjob.com',
    position: 'JS Full Stack Developer',
    description: 'No description yet',
    start: new Date('2018-06-11'),
    finish: 'Present',
    location: 'Barcelona, Spain',
    tools: ['React', 'Redux', 'Redux-Saga', 'Styled Components', 'Jenkins', 'Docker', 'AWS', 'Node'],
  },
  {
    image: 'image 2',
    companyName: 'Bigfinite Inc',
    companyLink: 'https://www.bigfinite.com/',
    position: 'JS Frontend Developer',
    description: `I was developing new user-facing features for the big web platform and providing bug-fixing support for already existing ones. Mostly my work was related to data representation of the different numeric and categoric AI and ML models and algorithms or other scientific things. 

    Technologies:
    Node.js | React.js | jQuery | HTML (Handlebars) | CSS (SASS) | D3 | Highcharts`,
    start: new Date('2017-04-18'),
    finish: new Date('2017-06-06'),
    location: 'Barcelona, Spain',
    tools: ['React', 'AWS', 'Node', 'jQuery', 'Highcharts'],
  },
  {
    image: 'image 3',
    companyName: 'Enixan Entertainment',
    companyLink: 'http://enixan.com/en/',
    position: 'Game Designer',
    description: `I have developed a new genre direction in the company (Match-3). There are a game created from my scratch and a prepared background for developing other similar projects with the developed game engine. 

    Main responsibilities:
    - market analysis of social and mobile games
    - development of the concept of games
    - development of the setting and narrative of games
    - elaboration of the economics (balance) of games 
    - elaboration of gameplay
    - level design
    - creation of game design documents
    - support of development, testing, consulting`,
    start: new Date('2015-09-07'),
    finish: new Date('2016-07-21'),
    location: 'Kyiv, Ukraine',
    tools: [],
  },
  {
    image: 'image 4',
    companyName: 'MBA Strategy',
    companyLink: 'http://www.mbastrategy.ru/',
    position: 'Math Consultant',
    description: `Teaching people how to prepare for the quantitative part of the GMAT/GRE.

    - individual lessons and group (4-9 students) lectures
    - online lessons via Skype
    - develop schemes of work, lesson plans and study materials
    - assign and grade class work, homework, tests and assignments
    - identify and select different instructional resources and methods to meet students' varying needs
    - organize and run educational events`,
    start: new Date('2013-01'),
    finish: new Date('2015-05'),
    location: 'Kyiv, Ukraine',
    tools: [],
  },
  {
    image: 'image 5',
    companyName: 'Icicle Seafoods',
    companyLink: 'http://www.icicleseafoods.com/',
    position: 'Cannery Worker',
    description: 'Work&Travel program which helped me to improve my English and teamwork skills. It made me more organized, more hardworking, more hardy.',
    start: new Date('2012-05'),
    finish: new Date('2012-09'),
    location: 'Alaska, USA',
    tools: [],
  },
];
