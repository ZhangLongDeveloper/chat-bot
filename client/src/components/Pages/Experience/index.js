import React from 'react';

import EmptyView from '../../Common/EmptyView';

import './styles.css';

const Experience = () => (
  <div>
    <EmptyView section="experience" />
  </div>
);

export default Experience;
