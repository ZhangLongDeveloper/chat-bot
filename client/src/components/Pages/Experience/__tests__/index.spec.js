import React from 'react';
import { create } from 'react-test-renderer';

import Experience from '../../Experience';

jest.mock('../../../Common/EmptyView', () => 'EmptyView');

const mockComponent = props => (
  <Experience {...props} />
);

describe('components/pages/<Experience />', () => {
  it('should render component', () => {
    const tree = create(mockComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
