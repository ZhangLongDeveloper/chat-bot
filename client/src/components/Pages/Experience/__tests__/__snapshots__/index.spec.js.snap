// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Pages/<Experience /> should render component 1`] = `
<div
  className="experiences-section-container"
>
  <h3
    className="experiences-section-title"
  >
    Experience
  </h3>
  <div
    className="experiences-wrapper"
  >
    <ExperienceItem
      companyLink="https://www.cornerjob.com"
      companyName="Cornerjob"
      description="Responsible for the whole FE of the platform,
      working with Webmanager and BO projects (React/Redux)
      and SEO-Landing project (Symfony/Twig/jQuery). Paying attention
      to UT, test coverage and CI/CD with Jest/Enzyme/Jenkins.
      Using mostly SCRUM methodologies and Kanban for some smaller projects.
    "
      id={5}
      image="test-file-stub"
      location="Barcelona, Spain"
      position="JS Full Stack Developer"
      start={2018-06-11T00:00:00.000Z}
      tools={
        Array [
          "React",
          "Redux",
          "Redux-Saga",
          "Jest",
          "Styled Components",
          "Symfony",
          "Jenkins",
          "Docker",
          "AWS",
          "Node",
        ]
      }
    />
    <ExperienceItem
      companyLink="https://www.bigfinite.com/"
      companyName="Bigfinite Inc"
      description="I was developing new user-facing features for
      the big web platform and providing bug-fixing support
      for already existing ones. Mostly my work was related to
      data representation of the different numeric and categoric
      AI and ML models and algorithms or other scientific things.
    "
      finish={2018-06-06T00:00:00.000Z}
      id={4}
      image="test-file-stub"
      location="Barcelona, Spain"
      position="JS Frontend Developer"
      start={2017-04-18T00:00:00.000Z}
      tools={
        Array [
          "React",
          "AWS",
          "Node",
          "jQuery",
          "Highcharts",
        ]
      }
    />
    <ExperienceItem
      companyLink="http://www.enixan.com/en/"
      companyName="Enixan Entertainment"
      description="I have developed a new genre direction in the
      company (Match-3). There are a game created from my scratch
      and a prepared background for developing other similar projects
      with the developed game engine.
    "
      finish={2016-07-21T00:00:00.000Z}
      id={3}
      image="test-file-stub"
      location="Kyiv, Ukraine"
      position="Game Designer"
      responsibilities={
        Array [
          "market analysis of social and mobile games",
          "development of the concept of games",
          "development of the setting and narrative of games",
          "elaboration of the economics (balance) of games",
          "elaboration of gameplay",
          "level design",
          "creation of game design documents",
          "support of development, testing, consulting",
        ]
      }
      start={2015-09-07T00:00:00.000Z}
    />
    <ExperienceItem
      companyLink="http://www.mbastrategy.ru/"
      companyName="MBA Strategy"
      description="Teaching people how to prepare for the quantitative part of the GMAT/GRE.

    "
      finish={2015-05-01T00:00:00.000Z}
      id={2}
      image="test-file-stub"
      location="Kyiv, Ukraine"
      position="Math Consultant"
      responsibilities={
        Array [
          "individual lessons and group (4-9 students) lectures",
          "online lessons via Skype",
          "develop schemes of work, lesson plans and study materials",
          "assign and grade class work, homework, tests and assignments",
          "identify and select different instructional resources and methods to meet students' varying needs",
          "organize and run educational events",
        ]
      }
      start={2013-01-01T00:00:00.000Z}
    />
    <ExperienceItem
      companyLink="http://www.icicleseafoods.com/"
      companyName="Icicle Seafoods"
      description="Work&Travel program which helped me to improve my English and teamwork skills. It made me more organized, more hardworking, more hardy."
      finish={2012-09-01T00:00:00.000Z}
      id={1}
      image="test-file-stub"
      location="Alaska, USA"
      position="Cannery Worker"
      start={2012-05-01T00:00:00.000Z}
    />
  </div>
</div>
`;
